trieste/utils/pareto.py

```diff
@@ -0,0 +1,309 @@
+from typing import Mapping
+
+import gpflow
+import numpy as np
+import tensorflow as tf
+
+from trieste.data import Dataset
+from trieste.type import TensorType
+
+jitter = gpflow.config.default_jitter()
+
+
+def atleast2d(tensor: tf.Tensor) -> tf.Tensor:
+    if tf.rank(tensor) <= 1:
+        return tf.expand_dims(tensor, axis=0)
+    return tensor
+
+
+def to_Dataset(datasets: Mapping[str, Dataset]) -> Dataset:
+
+    data_query_points = []
+    data_observations = []
+
+    for tag in datasets:
+        data_query_points.append(datasets[tag].query_points)
+        data_observations.append(datasets[tag].observations)
+
+    assert tf.math.reduce_all(
+        [tf.math.equal(input_data, data_query_points[0]) for input_data in data_query_points]
+    ), ValueError("Input data is not the same for each objective")
+
+    mo_dataset = Dataset(data_query_points[0], tf.concat(data_observations, axis=1))
+
+    return mo_dataset
```

Comment:
it looks like this is designed to allow users to express their multi-objective problem using multiple datasets. In particular, it's only used to reorganise function arguments. Is that right?

Reply:
kind of. it's used to reorganize multiple datasets into one `Datasets` with multiple outputs. we come up with this because trieste can't handle two different models with one Dataset with multiple y(output) CMIIW. [link](https://github.com/secondmind-labs/trieste/blob/0e8f34e87b9e1a0ec2ea6fd2fd857eff49c202ab/trieste/bayesian_optimizer.py#L233)